From: =?utf-8?q?Jonas_=C3=85dahl?= <jadahl@gmail.com>
Date: Thu, 14 Jan 2021 16:06:07 +0100
Subject: clutter/backend: Clear stage window pointer after removing weak ref

Otherwise when dispose() is called multiple times we spew warnings on
tear down.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1670>
---
 clutter/clutter/clutter-backend.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/clutter/clutter/clutter-backend.c b/clutter/clutter/clutter-backend.c
index 256c43a..546ac6c 100644
--- a/clutter/clutter/clutter-backend.c
+++ b/clutter/clutter/clutter-backend.c
@@ -98,6 +98,7 @@ clutter_backend_dispose (GObject *gobject)
     {
       g_object_remove_weak_pointer (G_OBJECT (backend->stage_window),
                                     (gpointer *) &backend->stage_window);
+      backend->stage_window = NULL;
     }
 
   G_OBJECT_CLASS (clutter_backend_parent_class)->dispose (gobject);
